pkg/cnb/project_descriptor.go

```diff
@@ -41,13 +41,22 @@ func ProcessProjectDescriptor(appDir, descriptorPath, platformDir string, logger
 	if err := processFiles(appDir, d.IO.Buildpacks.build); err != nil {
 		return err
 	}
-	return serializeEnvVars(d.IO.Buildpacks.Env, platformDir)
+
+	env := getEnvVariables(d)
+	return serializeEnvVars(env, platformDir)
 }
 
 func parseProjectDescriptor(file string, logger *log.Logger) (descriptorV2, error) {
 	var d descriptorV2
 	if _, err := toml.DecodeFile(file, &d); err != nil {
-		return descriptorV2{}, err
+		var dV2NoSpec descriptorV2NoSpec
+		if _, err2 := toml.DecodeFile(file, &dV2NoSpec); err2 != nil {
```

Comment:
Would it be easier to use the `encoding.TextUnmarshaler` interface to add logic to unmarshall the different types of `env` fields to avoid a whole different type?

Reply:
@matthewmcnew you mean to use [this](https://github.com/BurntSushi/toml#using-the-marshaler-and-encodingtextunmarshaler-interfaces)?

Reply:
Yeah!

Reply:
Thanks for the help @matthewmcnew. Done